### PR TITLE
update for clhep

### PR DESCRIPTION
**Pull request by lgarren**
Adam Stewart modified clhep package.py in my fork to use the newest spack packaging options.  I then modified it to find the latest clhep releases, which are in a different directory.  All the clhep 2.3.x.y releases are now available.

If you want to verify with a CLHEP maintainer, you're going to end up talking to me.

### CODE REVIEW

File: var/spack/repos/builtin/packages/clhep/package.py [hunk 1]
```diff
@@ -26,13 +26,23 @@
...
 
 
-class Clhep(Package):
+class Clhep(CMakePackage):
     """CLHEP is a C++ Class Library for High Energy Physics. """
     homepage = "http://proj-clhep.web.cern.ch/proj-clhep/"
     url      = "http://proj-clhep.web.cern.ch/proj-clhep/DISTRIBUTION/tarFiles/clhep-2.2.0.5.tgz"
-    list_url = "https://proj-clhep.web.cern.ch/proj-clhep/DISTRIBUTION/"
+    list_url = "https://proj-clhep.web.cern.ch/proj-clhep/"
+    list_depth = 3
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/clhep/package.py]: I just checked and I think a `list_depth` of 2 is sufficient. 1 works if you don't care about the 1.8 releases.

Comment (lgarren) [inline var/spack/repos/builtin/packages/clhep/package.py]: We definitely do not care about the 1.8 releases.  I verified that the depth of 1 works.  Thanks for catching that.

Comment (lgarren) [inline var/spack/repos/builtin/packages/clhep/package.py]: I note that if list_depth is not specified at all, no releases are found.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/clhep/package.py]: Yes, the default is 0, meaning Spack only searches that `list_url` for releases, it doesn't spider down through links.

**Comment by adamjstewart** [comment]
Just a quick flake8 bug to fix:
```
var/spack/repos/builtin/packages/clhep/package.py:75: [W293] blank line contains whitespace
```

**Comment by lgarren** [comment]
FIxed in commit 30b648035a9b834bf3f6752c989cb6bed0eda1f1 on my fork.  How to attach?

**Comment by adamjstewart** [comment]
It gets added to the PR automatically. Should be good to go after Travis passes.

**Pull request merged**